lib/api.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * REST API.
+ */
+
+// Internal packages:
+const package = require('../package.json')
+,   sink = require('./sink')
+,   util = require('./util')
+,   validator = require('./validator')
+;
+
+const Sink = sink.Sink
+,   version = package.version
+;
+
+/**
+ * Handle an API request: parse method and parameters, handle common errors and call the validator.
+ *
+ * @param {Object} req - HTTP request.
+ * @param {Object} res - HTTP result.
+ */
+
+const processRequest = function(req, res) {
+
+    if ('/api/version' === req.path) {
+        res.status(200).send(version);
+    }
+    else if ('/api/metadata' === req.path || '/api/validate' === req.path) {
+        const validate = ('/api/validate' === req.path);
+        var v
+        ,   v2
+        ,   options
+        ,   options2
+        ,   errors
+        ,   errors2
+        ,   handler
+        ,   handler2
+        ;
+        try {
+            options = util.processParams(
+                req.query,
+                undefined,
+                {required: (validate ? ['profile'] : []), forbidden: ['document', 'source']}
+            );
+        }
+        catch (err) {
+            return res.status(400).send(err.toString());
+        }
+        if (validate && 'auto' === options.profile) {
```

Comment:
Trying to validate an url with profile `auto` returns the profile only, eg.
`/api/validate?url=http://www.w3.org/TR/2016/WD-charmod-norm-20160407/&profile=auto` gives `WD`